cmd/api/src/daemons/datapipe/datapipe.go

```diff
@@ -209,8 +176,8 @@ func (s *Daemon) Start(ctx context.Context) {
 func (s *Daemon) deleteData() {
 	defer func() {
 		s.status.Update(model.DatapipeStatusIdle, false)
-		s.setDeletionRequested(false)
-		s.setAnalysisRequested(true)
+		_ = s.db.DeleteAnalysisRequest(s.ctx)
```

Comment:
Are `setDeletionRequested()` and it's replacement `DeleteAnalysisRequest()` doing the same thing? It seems like deletion and analysis are two separate processes

Reply:
They should be, `setDeletionRequested(false)` was resetting the in-memory state for deletion requests which is equivalent to deleting the analysis request row, (poorly named but what can you do, it's primarily for analysis).

It seemed the original intention of these requests (analysis | deletion) was to not ever have both active at the same time, this was formally guaranteed in this db implementation by storing the type instead of storing them separately as the in-memory approach originally did.